jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/StandardWarExplodedProcessor.java

```diff
@@ -0,0 +1,132 @@
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+class StandardWarExplodedProcessor implements ArtifactProcessor {
+
+  private final Path warPath;
+  private final Path targetExplodedWarRoot;
+  private final Integer warJavaVersion;
+  private final AbsoluteUnixPath appRoot;
+
+  /**
+   * Constructor for {@link StandardWarExplodedProcessor}.
+   *
+   * @param warPath path to WAR file
+   * @param targetExplodedWarRoot path to exploded-war root
+   * @param warJavaVersion war java version
+   * @param appRoot the absolute path of the app on the container
+   */
+  StandardWarExplodedProcessor(
+      Path warPath, Path targetExplodedWarRoot, Integer warJavaVersion, AbsoluteUnixPath appRoot) {
+    this.warPath = warPath;
+    this.targetExplodedWarRoot = targetExplodedWarRoot;
+    this.warJavaVersion = warJavaVersion;
+    this.appRoot = appRoot;
+  }
+
+  @Override
+  public List<FileEntriesLayer> createLayers() throws IOException {
+    // Clear the exploded-jar root first
+    if (Files.exists(targetExplodedWarRoot)) {
+      MoreFiles.deleteRecursively(targetExplodedWarRoot, RecursiveDeleteOption.ALLOW_INSECURE);
+    }
+
+    ZipUtil.unzip(warPath, targetExplodedWarRoot, true);
+    Predicate<Path> isFile = Files::isRegularFile;
+    Predicate<Path> isInWebInfLib =
+        path -> path.startsWith(targetExplodedWarRoot.resolve("WEB-INF").resolve("lib"));
+    Predicate<Path> isSnapshot = path -> path.getFileName().toString().contains("SNAPSHOT");
+
+    // Non-snapshot layer
+    Predicate<Path> isInWebInfLibAndIsNotSnapshot = isInWebInfLib.and(isSnapshot.negate());
+    Predicate<Path> nonSnapshotPredicate = isFile.and(isInWebInfLibAndIsNotSnapshot);
+    FileEntriesLayer nonSnapshotLayer =
+        ArtifactLayers.getDirectoryContentsAsLayer(
+            ArtifactLayers.DEPENDENCIES, targetExplodedWarRoot, nonSnapshotPredicate, appRoot);
+
+    // Snapshot layer
+    Predicate<Path> isInWebInfLibAndIsSnapshot = isInWebInfLib.and(isSnapshot);
+    Predicate<Path> snapshotPredicate = isFile.and(isInWebInfLibAndIsSnapshot);
+    FileEntriesLayer snapshotLayer =
+        ArtifactLayers.getDirectoryContentsAsLayer(
+            ArtifactLayers.SNAPSHOT_DEPENDENCIES,
+            targetExplodedWarRoot,
+            snapshotPredicate,
+            appRoot);
+
+    // Classes layer.
+    Predicate<Path> isClass = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isInWebInfClasses =
+        path -> path.startsWith(targetExplodedWarRoot.resolve("WEB-INF").resolve("classes"));
+    Predicate<Path> classesPredicate = isInWebInfClasses.and(isClass);
+    FileEntriesLayer classesLayer =
+        ArtifactLayers.getDirectoryContentsAsLayer(
+            ArtifactLayers.CLASSES, targetExplodedWarRoot, classesPredicate, appRoot);
+
+    // Resources layer.
+    Predicate<Path> resourcesPredicate = isInWebInfLib.or(isClass).negate();
```

Comment:
For consistency, maybe this?
```suggestion
    Predicate<Path> resourcesPredicate = isInWebInfLib.or(isClass).negate().and(isFile);
```
Or the other way around: maybe we consistently apply `isFile.and(...)` in the arguments to `getDirectoryContentsAsLayer` in all cases.